sdks/java/core/src/main/java/org/apache/beam/sdk/io/PubsubIO.java

```diff
@@ -1053,7 +1144,16 @@ private Bound(
        * @param <X> the type of the elements of the input {@link PCollection}
        */
       public <X> Bound<X> withCoder(Coder<X> coder) {
-        return new Bound<>(name, topic, timestampLabel, idLabel, coder);
+        return new Bound<>(name, topic, timestampLabel, idLabel, coder, null);
```

Comment:
I don't understand why the parse function is dropped here. Seems like something fundamental is missing.